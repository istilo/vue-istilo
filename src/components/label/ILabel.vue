<script setup lang="ts">
import { computed } from 'vue';

const props = defineProps<{
	/**
	 * A color modifier for label
	 */
	color: string;
}>();

const getClasses = computed(() => {
	return ['i-label', props.color ? `i-label-${props.color}` : ''];
});
</script>

<template>
	<div :class="getClasses">
		<slot />
	</div>
</template>
