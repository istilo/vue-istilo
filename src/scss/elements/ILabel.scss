@import '@/scss/helpers';
@import '@/scss/functions';
@import '@/scss/mixins';

// ============================================================
// Name:            Label
// Description:     Component to indicate important notes
//
// Component:       `#{$prefix}label`
//
// Modifier:        `#{$prefix}label-default`
//                  `#{$prefix}label-primary`
//                  `#{$prefix}label-secondary`
//                  `#{$prefix}label-success`
//                  `#{$prefix}label-warning`
//                  `#{$prefix}label-danger`
//
// ============================================================

// Variables
// ============================================================

$label-padding-vertical: margin(xsmall) !default;
$label-padding-horizontal: margin(small) !default;

$label-background: background-color(wrapper) !default;
$label-line-height: font(line-height) !default;
$label-font-size: font-size(small) !default;
$label-color: color(default) !default;
$label-border-radius: border(radius) !default;

$label-success-background: theme-color(success) !default;
$label-success-color: color(inverse) !default;
$label-warning-background: theme-color(warning) !default;
$label-warning-color: color(inverse) !default;
$label-danger-background: theme-color(danger) !default;
$label-danger-color: color(inverse) !default;

/* ============================================================
	Component: Label
   ============================================================ */

.#{$prefix}label {
	display: inline-block;
	padding: $label-padding-vertical $label-padding-horizontal;
	background: $label-background;
	line-height: $label-line-height;
	font-size: $label-font-size;
	color: $label-color;
	vertical-align: middle;
	white-space: nowrap;
	border-radius: $label-border-radius;

	font-weight: 400;
}

/* Color modifiers
   ============================================================ */

// @include create-modifiers(label);

.#{$prefix}label.#{$prefix}label-primary {
	background-color: rgba(theme-color(primary), 0.15);
	color: color(primary);
}

.#{$prefix}label.#{$prefix}label-secondary {
	background-color: rgba(theme-color(secondary), 0.15);
	color: color(secondary);
}

.#{$prefix}label.#{$prefix}label-success {
	background-color: rgba(theme-color(success), 0.15);
	color: color(success);
}

.#{$prefix}label.#{$prefix}label-warning {
	background-color: rgba(theme-color(warning), 0.15);
	color: color(warning);
}

.#{$prefix}label.#{$prefix}label-danger {
	background-color: rgba(theme-color(danger), 0.15);
	color: color(danger);
}
