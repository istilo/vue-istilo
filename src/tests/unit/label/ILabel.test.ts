import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { shallowMount, VueWrapper, enableAutoUnmount } from '@vue/test-utils';

import ILabel from '@/components/label/ILabel.vue';

enableAutoUnmount(afterEach);

describe('Istilo Label', () => {
	let wrapper: VueWrapper<any>;

	beforeEach(() => {
		wrapper = shallowMount(ILabel, {
			propsData: {
				color: '',
			},
			slots: {
				default: 'Label',
			},
		});

		expect(wrapper.exists()).toBe(true);
	});
	afterEach(() => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it('Should render correctly', async () => {
		expect(wrapper.element.tagName).toBe('DIV');
		expect(wrapper.classes()).toContain('i-label');

		expect(wrapper.text()).toBe('Label');
	});

	it('Should a color modifier class name', async () => {
		await wrapper.setProps({ color: 'primary' });

		expect(wrapper.classes()).toContain('i-label-primary');
	});
});
