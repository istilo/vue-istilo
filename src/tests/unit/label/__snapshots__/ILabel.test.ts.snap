// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Istilo Label > Should a color modifier class name 1`] = `"<div class=\\"i-label i-label-primary\\">Label</div>"`;

exports[`Istilo Label > Should render correctly 1`] = `"<div class=\\"i-label\\">Label</div>"`;
